### PR TITLE
Support boot disk interface

### PR DESCRIPTION
**Pull request by chentaozgoogle**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `boot_disk.interface` field to `google_compute_instance` resource
```

### CODE REVIEW

**Comment by chentaozgoogle** [comment]
hey @NickElliot , I have fixed all missing coverage shown in the Missing test report  beside the one for google_compute_instance_from_machine_image , which doesn't seem to be valid report.  The reason is that the `attached_disk` and `boot_disk`, where the  `interface` resides, cannot be configured in google_compute_instance_from_machine_image.  So please review . thanks!

**Review by NickElliot — DISMISSED** [review]
LGTM!

**Review by NickElliot — COMMENTED** [review]
Hi, TGC validation is having an issue due to not expecting the interface data. Looking into how that can be resolved but here is the error details for the time being:

```--- FAIL: TestFakeResourceData_getOkTypeObject (0.00s)
    fake_resource_data_test.go:242: 
        	Error Trace:	/home/runner/work/magic-modules/magic-modules/tfplan2cai/tfdata/fake_resource_data_test.go:242
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"device_name":"test-device_name", "disk_encryption_key_raw":"", "disk_encryption_key_sha256":"", "kms_key_self_link":"test-kms_key_self_link", "mode":"READ_ONLY", "source":"test-source"}
        	            	actual  : map[string]interface {}{"device_name":"test-device_name", "disk_encryption_key_raw":"", "disk_encryption_key_sha256":"", "interface":"", "kms_key_self_link":"test-kms_key_self_link", "mode":"READ_ONLY", "source":"test-source"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(map[string]interface {}) (len=6) {
        	            	+(map[string]interface {}) (len=7) {
        	            	  (string) (len=11) "device_name": (string) (len=16) "test-device_name",
        	            	@@ -4,2 +4,3 @@
        	            	  (string) (len=26) "disk_encryption_key_sha256": (string) "",
        	            	+ (string) (len=9) "interface": (string) "",
        	            	  (string) (len=17) "kms_key_self_link": (string) (len=22) "test-kms_key_self_link",
        	Test:       	TestFakeResourceData_getOkTypeObject
--- FAIL: TestFakeResourceData_getOknsetTypeObject (0.00s)
    fake_resource_data_test.go:317: 
        	Error Trace:	/home/runner/work/magic-modules/magic-modules/tfplan2cai/tfdata/fake_resource_data_test.go:317
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"device_name":"", "disk_encryption_key_raw":"", "disk_encryption_key_sha256":"", "kms_key_self_link":"", "mode":"", "source":""}
        	            	actual  : map[string]interface {}{"device_name":"", "disk_encryption_key_raw":"", "disk_encryption_key_sha256":"", "interface":"", "kms_key_self_link":"", "mode":"", "source":""}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,2 @@
        	            	-(map[string]interface {}) (len=6) {
        	            	+(map[string]interface {}) (len=7) {
        	            	  (string) (len=11) "device_name": (string) "",
        	            	@@ -4,2 +4,3 @@
        	            	  (string) (len=26) "disk_encryption_key_sha256": (string) "",
        	            	+ (string) (len=9) "interface": (string) "",
        	            	  (string) (len=17) "kms_key_self_link": (string) "",
        	Test:       	TestFakeResourceData_getOknsetTypeObject
FAIL
FAIL	github.com/GoogleCloudPlatform/terraform-google-conversion/v5/tfplan2cai/tfdata	0.044s
ok  	github.com/GoogleCloudPlatform/terraform-google-conversion/v5/tfplan2cai/tfplan	(cached)
FAIL
make: *** [Makefile:11: test] Error 1
Error: Process completed with exit code 2.```

**Review by NickElliot — CHANGES_REQUESTED** [review]
the new fields need to be added to `convertBootDisk` and `convertAttachedDisk` in `mmv1/third_party/cai2hcl/services/compute/compute_instance.go`

File: mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb [hunk 1]
```diff
@@ -206,6 +206,15 @@ func ResourceComputeInstance() *schema.Resource {
 							Description: `The RFC 4648 base64 encoded SHA-256 hash of the customer-supplied encryption key that protects this resource.`,
 						},
 
+						"interface": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"SCSI", "NVME"}, false),
+                                                        Computed:     false,
+							ForceNew:     false,
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb:214]:
```suggestion
```
These are default values, dont need to be explicitly declared

File: mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb [hunk 2]
```diff
@@ -206,6 +206,15 @@ func ResourceComputeInstance() *schema.Resource {
...
+							Description:  `The disk interface used for attaching this disk. One of SCSI or NVME. (This field is only used for specific cases, please don't specify this field without advice from Google.)`,
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance.go.erb:215]:
```suggestion
							Description:  `The disk interface used for attaching this disk. One of SCSI or NVME. (This field is shared with `attached_disk` and only used for specific cases, please don't specify this field without advice from Google.)`,
```
Explicitly mention this value is shared with Attached_disk

**Review by NickElliot — CHANGES_REQUESTED** [review]
last changes!

**Review by NickElliot — APPROVED** [review]
verified test failure is unrelated and original TGC failure is resolved -- LGTM!

**Pull request merged**